Library/Homebrew/formula_installer.rb

```diff
@@ -151,6 +151,28 @@ def check_install_sanity
     recursive_deps = formula.recursive_dependencies
     recursive_formulae = recursive_deps.map(&:to_formula)
 
+    recursive_dependencies = []
+    recursive_formulae.each do |dep|
+      dep_recursive_dependencies = dep.recursive_dependencies.map(&:to_s)
+      if dep_recursive_dependencies.include?(formula.name)
+        recursive_dependencies << "#{formula.full_name} depends on #{dep.full_name}"
+        recursive_dependencies << "#{dep.full_name} depends on #{formula.full_name}"
+      end
+    end
+
+    unless recursive_dependencies.empty?
```

Comment:
Is this just a shallow check to improve performance while next `if recursive_formulae.flat_map(&:recursive_dependencies).map(&:to_s).include?(formula.name)` is a deep check?